modules/programs/starship.nix

```diff
@@ -82,13 +82,25 @@ in
         https://starship.rs/advanced-config/#transientprompt-and-transientrightprompt-in-cmd
       '';
     };
+
+    configPath = mkOption {
+      type = lib.types.str;
+      default = ".config/starship.toml";
```

Comment:
```suggestion
      default = "${config.xdg.configHome}/starship.toml";
```

Reply:
Isn't this an absolute path to the user's home?

Reply:
https://github.com/nix-community/home-manager/blob/ae62fd8ad8347e6bb5b615057f39f33d595a1c47/modules/misc/xdg.nix#L70-L79 it's what we use to hold the actual xdg config home path in case someone changes its location.

Reply:
It is, but it gets handled properly from the activation script. We do it elsewhere in the codebase, too. 
https://github.com/nix-community/home-manager/blob/ae62fd8ad8347e6bb5b615057f39f33d595a1c47/modules/misc/xdg.nix#L70-L79

Reply:
Seems not to work

```
> cat /etc/profiles/per-user/rhoriguchi/etc/profile.d/hm-session-vars.sh
...
export STARSHIP_CONFIG="/home/rhoriguchi//home/rhoriguchi/.config/starship.toml"
...
```

Reply:
It works for the home file location, but it will screw up the env variable.

Reply:
Works for the `home.file` option. Had to use `lib.replaceStrings` for the env variable.

Reply:
This seems to break tests, since you need to have a state version. `if then` case for a certain `stateVersion`?